app/src/main/java/bg/zahov/app/data/local/RealmManager.kt

```diff
@@ -87,134 +106,126 @@ class RealmManager {
-                    updateTemplate = true
-                    fit = false
-                    automaticSync = true
-                }
+            findLatest(settings)?.apply {
+                language = Language.fromKey(LanguageKeys.ENGLISH)
+                units = Units.fromKey(UnitsKeys.METRIC)
+                soundEffects = true
+                theme = Theme.fromKey(ThemeKeys.DARK)
+                restTimer = 30
+                vibration = true
+                soundSettings = Sound.fromKey(SoundKeys.SOUND_1)
+                updateTemplate = true
+                enableWatch = false
+                automaticSync = true
             }
         }
     }
 
+    /**
+     * Updates a specified setting based on the title and new value provided.
+     *
+     * @param title The title of the setting to update.
+     * @param newValue The new value for the setting.
+     */
     suspend fun updateSetting(title: String, newValue: Any) = withRealm { realm ->
-        var settings: Settings? = null
-
-        getSettings().collect {
-            settings = it.obj
-        }
-
+        val settings = getSettingsOnce()
         realm.write {
-            settings?.let { coldSettings ->
-                findLatest(coldSettings)?.let {
-                    when (title) {
-                        LANGUAGE_SETTING -> {
-                            it.language = Language.fromKey((newValue as String))
-                        }
-
-                        UNIT_SETTING -> {
-                            it.units = Units.fromKey((newValue as String))
-                        }
-
-                        SOUND_EFFECTS_SETTING -> {
-                            it.soundEffects = (newValue as Boolean)
-                        }
-
-                        THEME_SETTING -> {
-                            it.theme = Theme.fromKey((newValue as String))
-                        }
-
-                        REST_TIMER_SETTING -> {
-                            it.restTimer = (newValue as Int)
-                        }
-
-                        VIBRATION_SETTING -> {
-                            it.vibration = (newValue as Boolean)
-                        }
-
-                        SOUND_SETTING -> {
-                            it.soundSettings = Sound.fromKey((newValue as String))
-                        }
-
-                        UPDATE_TEMPLATE_SETTING -> {
-                            it.updateTemplate = (newValue as Boolean)
-                        }
-
-                        FIT_SETTING -> {
-                            it.fit = (newValue as Boolean)
-                        }
-
-                        AUTOMATIC_SYNC_SETTING -> {
-                            it.automaticSync = (newValue as Boolean)
-                        }
-                    }
+            findLatest(settings)?.let {
+                when (title) {
+                    LANGUAGE_SETTING -> it.language = Language.fromKey(newValue as String)
```

Comment:
Don't use **as** for casting - it will throw exception if cannot cast the value. use **as?**

Reply:
fix it using a check newValue is Any